api/db-cdr/prep-tables/make-prep-survey.py

```diff
@@ -0,0 +1,455 @@
+                                "REGISTERED_ANSWER_SUPPRESSED" not in split_codes:
+                            # in the case that it doesn't have a code ex: Launched 5/30/2017 (PTSC) & 12/10/2019 (CE).
+                            if len(split_codes) > 1:
+                                long_code_to_short_code[split_codes[0].strip().lower()] = \
+                                    split_codes[1].strip().lower()
+
+                answer_suppression = {}
+                for item in field_annotation_array:
+                    split = item.split('=')
+                    if "CONTROLLED_ANSWER_SUPPRESSED" in split:
+                        answer_suppression.setdefault("CONTROLLED_ANSWER_SUPPRESSED", [])
+                        answer_suppression["CONTROLLED_ANSWER_SUPPRESSED"].append(split[1].lower())
+
+                    elif "REGISTERED_ANSWER_SUPPRESSED" in split:
+                        answer_suppression.setdefault("REGISTERED_ANSWER_SUPPRESSED", [])
+                        answer_suppression["REGISTERED_ANSWER_SUPPRESSED"].append(split[1].lower())
+
+                # for the first row of survey
+                if row_number == 1:
+                    survey_id_all = id_all
+                    write_row_all(all_writer, 0, question_code, question_label,
+                                  'SURVEY', None, None, item_flags, None)
+
+                    survey_id_controlled = id_controlled
+                    write_row(controlled_writer, 0, question_code, question_label,
+                              'SURVEY', None, None, item_flags, 'controlled', None)
+
+                    survey_id_registered = id_registered
+                    write_row(registered_writer, 0, question_code, question_label,
+                              'SURVEY', None, None, item_flags, 'registered', None)
+                    continue
+
+                ############ Topic ############
+                if len(row['Section Header']) > 0:
+                    header = row['Section Header']
+
+                    # find the new lines and index to get the header up to the new line
+                    # to do: need to figure out what to do with first topic in pers_med_his
+                    position = header.find("\n")
+                    if position == -1:
+                        header = row['Section Header'].replace("Thanks for your answers.", "")
+                    else:
+                        header = row['Section Header'][:position].replace("Thanks for your answers.", "")
+
+                    # get the concept codes for the topics and write topics into file
+                    topic_codes = query_topic_code(project, dataset, header.replace("'", "\\'"))
+                    topic_id_all = id_all
+                    topic_code = ''
+                    if not topic_codes:
+                        topic_code = topic_codes[0]
+                    write_row_all(all_writer, survey_id_all, topic_code, header, 'TOPIC', None, None, item_flags,
```

Comment:
the survey_id_*** variables are initialized in an IF statement above. Don't do that, because if the IF statement doesn't run, your variables are not defined